x-pack/plugin/autoscaling/src/main/java/org/elasticsearch/xpack/autoscaling/storage/ReactiveStorageDeciderService.java

```diff
@@ -375,7 +376,11 @@ private IndexMetadata indexMetadata(ShardRouting shard, RoutingAllocation alloca
             return allocation.metadata().getIndexSafe(shard.index());
         }
 
-        private Optional<String> highestPreferenceTier(List<String> preferredTiers, DiscoveryNodes unused) {
+        private Optional<String> highestPreferenceTier(
+            List<String> preferredTiers,
+            DiscoveryNodes unused,
+            DesiredNodes.ClusterMembers desiredNodes
```

Comment:
This parameter is never used, is that intentional?

Reply:
Yes, we use this method as a parameter in:
```
                    node -> DataTierAllocationDecider.shouldFilter(
                        indexMetadata,
                        node.node().getRoles(),
                        this::highestPreferenceTier,
                        allocation
                    )
```